estimagic/optimization/optimize.py

```diff
@@ -353,19 +387,24 @@ def _create_population(problem, algo_options, internal_params):
     return pop
 
 
-def _process_pygmo_results(pygmo_res, params, internal_params, constraints):
-    """Convert evolved population into json serializable dictionary.
+def _process_results(res, params, internal_params, constraints, origin):
+    """Convert optimization results into json serializable dictionary.
 
-    Todo:
-        - implement this function.
+    Args:
+        res: Result from numerical optimizer.
+        params (DataFrame): See :ref:`params_df`.
+        internal_params (DataFrame): See :ref:`params_df`.
+        constraints (list): constraints for the optimization
+        origin (str): takes the values "pygmo", "nlopt", "scipy"
 
     """
-    x = pygmo_res.champion_x
+    if origin == "scipy":
+        x = res["x"]
+        f = res["fun"]
+    elif origin in ["pygmo", "nlopt"]:
+        x = res.champion_x
+        f = res.champion_f
     params = _params_sr_from_x(x, internal_params, constraints, params)
 
-    res_dict = {
-        "x": params.to_numpy().tolist(),
-        "internal_x": x.tolist(),
-        "f": pygmo_res.champion_f,
-    }
+    res_dict = {"x": params.to_numpy().tolist(), "internal_x": x.tolist(), "f": f}
```

Comment:
res_dict[f] has different types depending on the optimizer used. Using scipy (TNC), a np.array of shape (1,) is returned. If nlopt (bobyqa) is used, a float is returned. The later is probably more desirable.

Reply:
Resolved in fb49447181d310b40a25b48455ec1abea31905db

Reply:
Thanks for solving.